security/pkg/nodeagent/caclient/interface/iclient.go

```diff
@@ -0,0 +1,25 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package iclient
+
+import (
+	"context"
+)
+
+// Client interface defines the clients need to implement to talk to CA for CSR.
+type Client interface {
```

Comment:
Please don't define the interface in separated file.
Golang suggests defining it when it's used:
https://github.com/golang/go/wiki/CodeReviewComments#interfaces

Reply:
the interface is used in both caclient/client.go and caclient/providers/google/client.go, have to put in separated file(not in caclient/client.go) to avoid circular dependency.